### PR TITLE
HBASE-24480: Deflake TestRSGroupsBasics#testClearDeadServers

### PR DESCRIPTION
**Pull request by bharathv**
More details about the flakiness in the jira comments.

### CODE REVIEW

File: hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java [hunk 1]
```diff
@@ -385,7 +386,7 @@ public RSGroupInfoManager getRSGroupInfoManager() throws IOException {
   }
 
   @Override
-  public void removeServers(Set<Address> servers) throws IOException {
+  public void removeServers(@Nullable Set<Address> servers) throws IOException {
```
Comment (Reidddddd) [inline hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java:389]: I don't mean annotation `Nullable` is not good, but HBase doesn't use it around the codebases. So here makes me feel not unified, what about comment the nullable at the method @ parma description.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 34s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 45s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 18s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   0m 23s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  0s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   1m  9s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  7s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 52s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 23s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   0m 23s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 16s |  hbase-rsgroup: The patch generated 2 new + 2 unchanged - 0 fixed = 4 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 47s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 32s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 16s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   0m 57s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  11m 51s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 22s |  The patch does not generate ASF License warnings.  |
|  |   |  43m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1821 |
| JIRA Issue | HBASE-24480 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux f3d73612efcf 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1821/out/precommit/personality/provided.sh |
| git revision | branch-1 / 3c13884 |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/1/artifact/out/diff-checkstyle-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/1/testReport/ |
| Max. process+thread count | 1607 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/1/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

File: hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java (hunk 1 above)
Comment (bharathv) [reply hbase-rsgroup/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupAdminServer.java]: Fair point, it does look awkward when very few places in the codebase use that. Removed it.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   9m 45s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 18s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   0m 23s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   3m  4s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 32s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 20s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   1m  8s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m  6s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   1m 54s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 18s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   0m 18s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 23s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   0m 23s |  the patch passed  |
| -1 :x: |  checkstyle  |   0m 17s |  hbase-rsgroup: The patch generated 2 new + 2 unchanged - 0 fixed = 4 total (was 2)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   2m 46s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   4m 29s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 14s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   1m  0s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  12m  7s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 21s |  The patch does not generate ASF License warnings.  |
|  |   |  44m 10s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1821 |
| JIRA Issue | HBASE-24480 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux ded2039ad577 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1821/out/precommit/personality/provided.sh |
| git revision | branch-1 / 6bca1dd |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/2/artifact/out/diff-checkstyle-hbase-rsgroup.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/2/testReport/ |
| Max. process+thread count | 1609 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/2/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m  0s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| +1 :green_heart: |  test4tests  |   0m  0s |  The patch appears to include 1 new or modified test files.  |
||| _ branch-1 Compile Tests _ |
| +1 :green_heart: |  mvninstall  |  10m 21s |  branch-1 passed  |
| +1 :green_heart: |  compile  |   0m 22s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  compile  |   0m 29s |  branch-1 passed with JDK v1.7.0_262  |
| +1 :green_heart: |  checkstyle  |   0m 31s |  branch-1 passed  |
| +1 :green_heart: |  shadedjars  |   4m  3s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 35s |  branch-1 passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 21s |  branch-1 passed with JDK v1.7.0_262  |
| +0 :ok: |  spotbugs  |   1m 23s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   1m 18s |  branch-1 passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   2m 48s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 22s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javac  |   0m 22s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 30s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  javac  |   0m 30s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   3m 52s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |   5m 58s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2.  |
| +1 :green_heart: |  javadoc  |   0m 19s |  the patch passed with JDK v1.8.0_252  |
| +1 :green_heart: |  javadoc  |   0m 25s |  the patch passed with JDK v1.7.0_262  |
| +1 :green_heart: |  findbugs  |   1m 26s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |  13m 10s |  hbase-rsgroup in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 24s |  The patch does not generate ASF License warnings.  |
|  |   |  53m 28s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.10 Server=19.03.10 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/1821 |
| JIRA Issue | HBASE-24480 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux febcea803291 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/Base-PreCommit-GitHub-PR_PR-1821/out/precommit/personality/provided.sh |
| git revision | branch-1 / 6bca1dd |
| Default Java | 1.7.0_262 |
| Multi-JDK versions | /usr/lib/jvm/zulu-8-amd64:1.8.0_252 /usr/lib/jvm/zulu-7-amd64:1.7.0_262 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/3/testReport/ |
| Max. process+thread count | 1605 (vs. ulimit of 10000) |
| modules | C: hbase-rsgroup U: hbase-rsgroup |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-1821/3/console |
| versions | git=1.9.1 maven=3.0.5 findbugs=3.0.1 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by virajjasani** [comment]
Looks like you removed them already :)

**Pull request merged**